[BUG] - CodeBuild for lake-admin regression stuck with k8s dynamic client unauthorized error.

**Describe the bug**
PyTest based lake-admin regression code build stage hangs for EKS cluster connection via DynamicClient. 

Based on the availability of DynamicClient, OrbitJob status is checked for finalizing the PyTest status. 

Same code from developer local machine runs as expected to PASS or FAIL the PyTest. 

**To Reproduce**
Steps to reproduce the behavior:
Run lake-admin regression state in dev account.

**Expected behavior**
The PyTest running inside the codebuild has to check the OrbitJob status periodically and finalize the status of the PyTest.

**Screenshots**


WARNING  kubetest:utils.py:130 got api exception while waiting: 401
--
953 | Reason: Unauthorized
954 | HTTP response headers: HTTPHeaderDict({'Audit-Id': '74627bd8-d6c3-42d1-8fb1-19b9d7bc0550', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'Date': 'Wed, 06 Oct 2021 03:47:37 GMT', 'Content-Length': '129'})
955 | HTTP response body: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Unauthorized","reason":"Unauthorized","code":401}\n'
956 | Original traceback:
957 | File "/codebuild/output/src916/src/s3/00/.venv/lib/python3.8/site-packages/kubernetes/dynamic/client.py", line 55, in inner
958 | resp = func(self, *args, **kwargs)
959 |  
960 | File "/codebuild/output/src916/src/s3/00/.venv/lib/python3.8/site-packages/kubernetes/dynamic/client.py", line 237, in request
961 | return self.client.call_api(
962 |  
963 | File "/codebuild/output/src916/src/s3/00/.venv/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 348, in call_api
964 | return self.__call_api(resource_path, method,
965 |  
966 | File "/codebuild/output/src916/src/s3/00/.venv/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 180, in __call_api
967 | response_data = self.request(
968 |  
969 | File "/codebuild/output/src916/src/s3/00/.venv/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 373, in request
970 | return self.rest_client.GET(url,
971 |  
972 | File "/codebuild/output/src916/src/s3/00/.venv/lib/python3.8/site-packages/kubernetes/client/rest.py", line 239, in GET
973 | return self.request("GET", url,
974 |  
975 | File "/codebuild/output/src916/src/s3/00/.venv/lib/python3.8/site-packages/kubernetes/client/rest.py", line 233, in request
976 | raise ApiException(http_resp=r)
977 | - generated xml file: /codebuild/output/src916/src/s3/01/test/regressions/regression_report.xml -
978 | =========================== short test summary info ============================
979 | FAILED test_lake_admin.py::test_lakeadmin_2_image_with_apps - TimeoutError: t...
980 | !!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
981 | !!!!!!!!!!!! xdist.dsession.Interrupted: stopping after 1 failures !!!!!!!!!!!!!
982 | =================== 1 failed, 2 passed in 7204.35s (2:00:04) ===================



**Desktop (please complete the following information):**
 - OS: [e.g. iOS]
 - Browser [e.g. chrome, safari]
 - Version [e.g. 22]

**Additional context**
Add any other context about the problem here.

--- a/test/regressions/common_utils.py
+++ b/test/regressions/common_utils.py
@@ -12,6 +12,81 @@
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
 
+import time
+import logging
+from typing import Union
+from kubernetes.client.rest import ApiException
+from kubernetes import config as k8_config
+from kubetest import condition
 
 JOB_COMPLETION_STATUS = "Complete"
 JOB_FAILED_STATUS = "Failed"
+
+log = logging.getLogger("kubetest")
+
+
+def wait_for_custom_condition(
+    condition: condition.Condition,
+    timeout: int = None,
+    interval: Union[int, float] = 1,
+    fail_on_api_error: bool = True,
+) -> None:
+    """Wait for a condition to be met.
+
+    Args:
+        condition: The Condition to wait for.
+        timeout: The maximum time to wait, in seconds, for the condition to be met.
+            If unspecified, this function will wait indefinitely. If specified and
+            the timeout is met or exceeded, a TimeoutError will be raised.
+        interval: The time, in seconds, to wait before re-checking the condition.
+        fail_on_api_error: Fail the condition checks if a Kubernetes API error is
+            incurred. An API error can be raised for a number of reasons, including
+            a Pod being restarted and temporarily unavailable. Disabling this will
+            cause those errors to be ignored, allowing the check to continue until
+            timeout or resolution. (default: True).
+
+    Raises:
+        TimeoutError: The specified timeout was exceeded.
+    """
+    log.info(f"waiting for condition: {condition}")
+
+    # define the maximum time to wait. once this is met, we should
+    # stop waiting.
+    max_time = None
+    if timeout is not None:
+        max_time = time.time() + timeout
+
+    # start the wait block
+    start = time.time()
+    count = 0
+    retry_count = 3
+    while True:
+        if max_time and time.time() >= max_time:
+            raise TimeoutError(
+                f"timed out ({timeout}s) while waiting for condition {condition}"
+            )
+
+        # check if the condition is met and break out if it is
+        try:
+            if condition.check():
+                break
+        except ApiException as e:
+            log.warning(f"got api exception while waiting: {e}")
+            if count < retry_count:
+                if e.get('reason') == 'Unauthorized':
+                    k8_config.load_kube_config()
+                    log.info("loading the kubeconfig during Unauthorized exception and retrying")
+                    count+=1
+                    continue
+                else:
+                    raise
+            else:
+                log.info(f"retried loading kubeconfig {retry_count} times")
+                raise
+
+        # if the condition is not met, sleep for the interval
+        # to re-check later
+        time.sleep(interval)
+
+    end = time.time()
+    log.info(f"wait completed (total={end-start}s) {condition}")

--- a/test/regressions/custom_resources.py
+++ b/test/regressions/custom_resources.py
@@ -18,12 +18,11 @@ from typing import Any, Dict, List, cast, Optional, Union
 import yaml
 from kubernetes import dynamic
 from kubernetes.client import V1DeleteOptions, V1Status, api_client
-from kubernetes import config as k8_config
 from kubetest.manifest import render
 from kubetest.objects import ApiObject
 from kubetest.client import TestClient
-from kubetest import condition, utils
-from common_utils import JOB_COMPLETION_STATUS, JOB_FAILED_STATUS
+from kubetest import condition
+from common_utils import JOB_COMPLETION_STATUS, JOB_FAILED_STATUS, wait_for_custom_condition
 
 # This elminates some collection warnings for all tests using the TestClient
 TestClient.__test__ = False
@@ -76,7 +75,6 @@ class CustomApiObject(ApiObject):  # type: ignore
     @property
     def api_client(self) -> dynamic.DynamicClient:
         if self._api_client is None:
-            k8_config.load_kube_config()
             c = self.api_clients.get(self.version)
             # If we didn't find the client in the api_clients dict, use the
             # preferred version.
@@ -247,7 +245,7 @@ class OrbitJobCustomApiObject(CustomApiObject):
             self.is_complete,
         )
         # Wait until Orbit job completes
-        utils.wait_for_condition(
+        wait_for_custom_condition(
             condition=job_complete_condition,
             timeout=timeout,
             interval=interval,

